scripts/generate-sql.js

```diff
@@ -273,17 +275,101 @@ FROM "AllPseudos_${projectNameChunked.join('_')}";
 -- created in the 0.code-sets.sql file
 DROP TABLE IF EXISTS ${config.PROJECT_SPECIFIC_SCHEMA_FOR_DATA}.${tableName};
 CREATE TABLE ${config.PROJECT_SPECIFIC_SCHEMA_FOR_DATA}.${tableName} AS
-SELECT ${config.PROJECT_SPECIFIC_SCHEMA_PRIVATE_TO_RDES}.gm_pseudo_hash_${projectNameChunked.join(
+SELECT ${config.PROJECT_SPECIFIC_SCHEMA_PRIVATE_TO_RDES}.gm_pseudo_hash_${projectNameChunked.join('_')}("GmPseudo") AS "PatientID",
+	* EXCLUDE "GmPseudo"
+FROM ${config.PROJECT_SPECIFIC_SCHEMA_PRIVATE_TO_RDES}."${tableNameNoQuotes}_WITH_PSEUDO_IDS";`;
+
+    sql =
+      sql.substring(0, indexOfOutputTable) +
+      replacedSql +
+      finalSql +
+      sql.substring(indexOfFinalSemiColon + 1);
+  } else if (outputTableMatchedCohortMatch) {
+    isMatch = true;
+    let needSemiColon = false;
+
+    const indexOfOutputTable = sql.indexOf('{{create-output-table-matched-cohort');
+    let indexOfFinalSemiColon = sql.indexOf(';', indexOfOutputTable);
+    if (indexOfFinalSemiColon < 0) {
+      needSemiColon = true;
+      indexOfFinalSemiColon = sql.length - 1;
+    }
+
+    const [, tableName] = outputTableMatchedCohortMatch;
+    const tableNameNoQuotes = tableName.match(/^"?([^"]+)"?$/)[1];
+
+    const replacedSql =
+      sql.substring(indexOfOutputTable, indexOfFinalSemiColon + 1).replace(
+        outputTableMatchedCohortRegex,
+        `
+-- ... processing ${outputTableMatchedCohortMatch[0].replace(/\{/g, '[').replace(/\}/g, ']')} ... 
+-- ... Need to create an output table called ${tableName} and replace 
+-- ... the GmPseudo column with a study-specific random patient id.
+
+-- First we create a table in an area only visible to the RDEs which contains
+-- the GmPseudos. THESE CANNOT BE RELEASED TO END USERS.
+DROP TABLE IF EXISTS ${
+          config.PROJECT_SPECIFIC_SCHEMA_PRIVATE_TO_RDES
+        }."${tableNameNoQuotes}_WITH_PSEUDO_IDS";
+CREATE TABLE ${
+          config.PROJECT_SPECIFIC_SCHEMA_PRIVATE_TO_RDES
+        }."${tableNameNoQuotes}_WITH_PSEUDO_IDS" AS`
+      ) + (needSemiColon ? ';' : '');
+
+    const finalSql = `
+
+-- Then we check to see if there are any new GmPseudo ids. We do this by making a temp table 
+-- of all "new" GmPseudo ids from either the main column or the matched column. I.e. any GmPseudo ids that 
+-- we've already got a unique id for for this study are excluded
+
+DROP TABLE IF EXISTS "AllPseudos_${projectNameChunked.join('_')}";
+CREATE TEMPORARY TABLE "AllPseudos_${projectNameChunked.join('_')}" AS
+SELECT DISTINCT "GmPseudo" FROM ${
```

Comment:
I'm sure it's fine but in situations like this where I'm using more than one UNION/EXCEPT/INTERSECT statement, I tend to use brackets. E.g. `(SELECT ... UNION SELECT ...) EXCEPT SELECT...`. This is because I'm nervous that the database will do the operations in the wrong order (e..g the EXCEPT first, and then the UNION rather than vice versa) leading to the wrong result. It also perhaps makes it easier for someone reading the code to work out what the intent is.